src/odemis/driver/actuator.py

```diff
@@ -1366,3 +1366,510 @@ def terminate(self):
+        # axis names of axes of children
+        self._caxes_map = caxes_map
+        self._atol = atol
+
+        for i, (c, ac) in enumerate(zip(self._children, self._caxes_map)):
+            if ac not in c.axes:
+                raise ValueError("Child %s has no axis named %s" % (c.name, ac))
+
+            if hasattr(c.axes[ac], "range"):
+                mn, mx = c.axes[ac].range
+                for key, pos in self._positions.items():
+                    if not mn <= pos[i] <= mx:
+                        raise ValueError("Position %s with key %s is out of range for children." % (pos[i], key))
+
+            elif hasattr(c.axes[ac], "choices"):
+                for pos in self._positions.values():
+                    if pos[i] not in c.axes[ac].choices:
+                        raise ValueError("Position %s is not in range of choices for children." % pos[i])
+
+        axes = {axis_name: model.Axis(choices=set(positions.keys() + [fallback]))}
+
+        # this set ._axes and ._children
+        model.Actuator.__init__(self, name, role, axes=axes, children=children, **kwargs)
+
+        # will take care of executing axis move asynchronously
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
+
+        # create position VA and subscribe to position
+        self.position = model.VigilantAttribute({}, readonly=True)
+        for c in self._children:
+            c.position.subscribe(self._updatePosition)
+
+        # update position VA based on current positions of children axes
+        self._updatePosition()
+
+        # check if children axes can be referenced, create referenced VA, subscribe to referenced
+        self._children_refd = [model.hasVA(child, "referenced") and caxis in child.referenced.value
+                               for child, caxis in zip(self._children, self._caxes_map)]
+        if any(self._children_refd):
+            # whether the axes are referenced
+            self.referenced = model.VigilantAttribute({}, readonly=True)
+            for c in [c for c, r in zip(self._children, self._children_refd) if r]:
+                c.referenced.subscribe(self._updateReferenced)
+
+            self._updateReferenced()
+
+    def _readPositionfromChildren(self):
+        """"""
+        pos_children = [self._children[0].position.value[self._caxes_map[0]],
+                        self._children[1].position.value[self._caxes_map[1]]]
+
+        # check whether children positions are in consistency with combined position allowed
+        pos_matching = map(
+            lambda pos: util.almost_equal(pos_children[0], pos[0], atol=self._atol[0], rtol=0) &
+                        util.almost_equal(pos_children[1], pos[1], atol=self._atol[1], rtol=0)
+                        if isinstance(pos[0], float) and isinstance(pos[1], float) else False,
+                        self._positions.values())
+
+        return pos_matching, pos_children
```

Comment:
Done! I unsubscribe now twice in case I do not find a matching position as I call MoveAbs within the referencing. Is it a problem to unsubscribe twice?